lib/spack/docs/binary_caches.rst

```diff
@@ -1,108 +1,132 @@
 .. _binary_caches:
 
+============
 Build caches
 ============
 
 Some sites may encourage users to set up their own test environments
-before carrying out central installations, or some users prefer to set
+before carrying out central installations, or some users may prefer to set
 up these environments on their own motivation. To reduce the load of
 recompiling otherwise identical package specs in different installations,
-installed packages can be put into build cache tarballs, uploaded to 
-your spack mirror and then downloaded and installed by others.
+installed packages can be put into build cache tarballs, uploaded to
+your Spack mirror and then downloaded and installed by others.
 
 
+--------------------------
 Creating build cache files
 --------------------------
 
 A compressed tarball of an installed package is created. Tarballs are created
 for all of its link and run dependency packages as well. Compressed tarballs are
-signed with gpg and signature and tarball and put in a ".spack" file. Optionally
-, the rpaths ( and ids and deps on macOS ) can be changed to paths relative to 
-the spack install tree before the tarball is created.
+signed with gpg and signature and tarball and put in a ``.spack`` file. Optionally,
+the rpaths (and ids and deps on macOS) can be changed to paths relative to
+the Spack install tree before the tarball is created.
 
 Build caches are created via:
 
-.. code-block:: sh
+.. code-block:: console
 
-   $ spack buildcache create 
+   $ spack buildcache create
 
 
+---------------------------------------
 Finding or installing build cache files
 ---------------------------------------
 
-To find build caches or install build caches, a spack mirror must be configured
-with
- 
-``spack mirror add <name> <url>``. 
+To find build caches or install build caches, a Spack mirror must be configured
+with:
+
+.. code-block:: console
+
+   $ spack mirror add <name> <url>
 
-Build caches are found via: 
+Build caches are found via:
 
-.. code-block:: sh
+.. code-block:: console
 
    $ spack buildcache list
 
 Build caches are installed via:
 
-.. code-block:: sh
+.. code-block:: console
 
-   $ spack buildcache install 
-   
+   $ spack buildcache install
 
+
+----------
 Relocation
 ----------
 
-Initial build and later installation do not necessarily happen at the same 
-location. Spack provides a relocation capability and corrects for RPATHs and 
-non-relocatable scripts. However, many packages compile paths into binary 
-artificats directly. In such cases, the build instructions of this package would
+Initial build and later installation do not necessarily happen at the same
+location. Spack provides a relocation capability and corrects for RPATHs and
+non-relocatable scripts. However, many packages compile paths into binary
+artifacts directly. In such cases, the build instructions of this package would
 need to be adjusted for better re-locatability.
 
+.. _cmd-spack-buildcache:
```

Comment:
This adds a link in the [Command Index](http://spack.readthedocs.io/en/latest/command_index.html).